packages/react-sdk/src/state/useActiveWhiteboardMembers.ts

```diff
@@ -24,20 +28,35 @@ export type ActiveWhiteboardMember = {
 
 export function useActiveWhiteboardMembers(): ActiveWhiteboardMember[] {
   const statistics = useActiveWhiteboardInstanceStatistics();
+  const { data: rtcMembers } = useGetRtcMembersQuery();
 
   return useMemo(() => {
     const activeWhiteboardMembers = new Map<string, ActiveWhiteboardMember>();
 
-    Object.values(statistics.communicationChannel.peerConnections).forEach(
-      (p) => {
-        if (isPeerConnected(p)) {
-          activeWhiteboardMembers.set(p.remoteUserId, {
-            userId: p.remoteUserId,
-          });
-        }
-      },
-    );
+    if (matrixRtcMode) {
+      const allRtcMembers = rtcMembers
+        ? selectRtcMembers(rtcMembers).filter((m) =>
+            isRTCSessionNotExpired(m.content),
+          )
+        : undefined;
 
-    return Array.from(activeWhiteboardMembers.values());
-  }, [statistics]);
+      allRtcMembers?.forEach((m) => {
+        activeWhiteboardMembers.set(m.sender, {
+          userId: m.sender,
+        });
+      });
+      return Array.from(activeWhiteboardMembers.values());
+    } else {
+      Object.values(statistics.communicationChannel.peerConnections).forEach(
+        (p) => {
+          if (isPeerConnected(p)) {
+            activeWhiteboardMembers.set(p.remoteUserId, {
+              userId: p.remoteUserId,
+            });
+          }
+        },
+      );
+      return Array.from(activeWhiteboardMembers.values());
```

Comment:
I think this like with Array creation is kind of duplicated for other if branch. I suggest to move it after the if block.